python/cudf/cudf/core/buffer/exposure_tracked_buffer.py

```diff
@@ -0,0 +1,311 @@
+# Copyright (c) 2020-2023, NVIDIA CORPORATION.
+
+from __future__ import annotations
+
+import weakref
+from typing import (
+    Any,
+    Container,
+    Literal,
+    Mapping,
+    Optional,
+    Type,
+    TypeVar,
+    cast,
+)
+
+from typing_extensions import Self
+
+import cudf
+from cudf.core.buffer.buffer import Buffer, get_ptr_and_size
+from cudf.utils.string import format_bytes
+
+T = TypeVar("T", bound="ExposureTrackedBuffer")
+
+
+def get_owner(data, klass: Type[T]) -> Optional[T]:
+    """Get the owner of `data`, if any exist
+
+    Search through the stack of data owners in order to find an
+    owner of type `klass` (not subclasses).
+
+    Parameters
+    ----------
+    data
+        The data object
+
+    Return
+    ------
+    klass or None
+        The owner of `data` if `klass` or None.
+    """
+
+    if type(data) is klass:
+        return data
+    if hasattr(data, "owner"):
+        return get_owner(data.owner, klass)
+    return None
+
+
+def as_exposure_tracked_buffer(
```

Comment:
Can we define this in utils.py since that's where as_buffer is defined?

Reply:
I like to wait doing this until `SpillableBuffer` also becomes a `ExposureTrackedBuffer`.
At the moment, `as_buffer()`, imports spillable and exposure tracked buffers as orthogonal types:
```python
from cudf.core.buffer.exposure_tracked_buffer import as_exposure_tracked_buffer
from cudf.core.buffer.spillable_buffer import SpillLock, as_spillable_buffer
```

Reply:
OK got it, that's fine with me.